### PR TITLE
refactor(server): remove spawning of 2nd tokio runtime

### LINKED ISSUE #4448: server: remove spawning of 2nd tokio runtime
## Description

When the `libp2p-server` was moved into the repository, its main runtime was changed to `tokio` from `async-std`. What we missed is that we spawn another `tokio` runtime for the metrics server: https://github.com/libp2p/rust-libp2p/blob/master/misc/server/src/http_service.rs#L42

## Motivation

Simpler code.

## Current Implementation

Two tokio runtimes are being spawned.

### PR DESCRIPTION
**Pull request by chirag-bgh**
## Description
Closes #4448.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: misc/server/src/http_service.rs [hunk 1]
```diff
@@ -38,21 +38,16 @@ pub(crate) async fn metrics_server(
...
-    })
+    let server = Server::bind(&addr).serve(MakeMetricService::new(registry, metrics_path.clone()));
+    info!(
+        "Metrics server on http://{}{}",
+        server.local_addr(),
+        metrics_path
+    );
+    if let Err(e) = server.await {
+        error!("server error: {}", e);
+    }
```
Comment (thomaseizinger) [inline misc/server/src/http_service.rs]: I think we can now just bubble this one up with `?`!

**Review by thomaseizinger — COMMENTED** [review]
The metrics server is spawned onto a new thread using the `futures` executor inside `main.rs`. Can you also fix that please? I think we should just be able to spawn it as a task :)

File: misc/server/src/http_service.rs (hunk 1 above)
Comment (chirag-bgh) [reply misc/server/src/http_service.rs]: But can we do it? Because `hyper::Error` is defined in external crate.

Comment (thomaseizinger) [reply misc/server/src/http_service.rs]: Why does that matter? We should still be able to spawn it as a task onto the runtime?

Comment (thomaseizinger) [reply misc/server/src/http_service.rs]: I don't really care how we organise the code, I would just like to only use the tokio runtime without additional threads / executors :)

**Comment by thomaseizinger** [comment]
Friendly ping @chirag-bgh, let me know if you need any more input on this :)

**Comment by chirag-bgh** [comment]
> Friendly ping @chirag-bgh, let me know if you need any more input on this :)

ahh so sorry abt this. will get this cleared today.

**Comment by thomaseizinger** [comment]
I pushed some follow-up commits, thank you for getting this started!

**Review by thomaseizinger — APPROVED** [review]

**Review by mxinden — COMMENTED** [review]
Thank you for the clean-up!

**Pull request merged**